pytest-backend/exasol/pytest_backend/__init__.py

```diff
@@ -0,0 +1,218 @@
+    if use_onprem:
+        return {
+            'dsn': f'{exasol_config.host}:{exasol_config.port}',
+            'user': exasol_config.username,
+            'password': exasol_config.password,
+            'websocket_sslopt': {'cert_reqs': ssl.CERT_NONE}
+        }
+    return {}
+
+
+@pytest.fixture(scope="session")
+def backend_aware_saas_database_params(use_saas,
+                                       saas_host,
+                                       saas_pat,
+                                       saas_account_id,
+                                       backend_aware_saas_database_id) -> dict[str, Any]:
+    if use_saas:
+        conn_params = get_connection_params(host=saas_host,
+                                            account_id=saas_account_id,
+                                            database_id=backend_aware_saas_database_id,
+                                            pat=saas_pat)
+        conn_params['websocket_sslopt'] = {'cert_reqs': ssl.CERT_NONE}
+        return conn_params
+    return {}
+
+
+@pytest.fixture(scope="session")
+def backend_aware_onprem_bucketfs_params(use_onprem,
+                                         backend_aware_onprem_database,
+                                         bucketfs_config) -> dict[str, Any]:
+    if use_onprem:
+        return {
+            'backend': _BACKEND_ONPREM,
+            'url': bucketfs_config.url,
+            'username': bucketfs_config.username,
+            'password': bucketfs_config.password,
+            'service_name': 'bfsdefault',
+            'bucket_name': 'default',
+            'verify': False
+        }
+    return {}
+
+
+@pytest.fixture(scope="session")
+def backend_aware_saas_bucketfs_params(use_saas,
+                                       saas_host,
+                                       saas_pat,
+                                       saas_account_id,
+                                       backend_aware_saas_database_id) -> dict[str, Any]:
+    if use_saas:
+        return {
+            'backend': _BACKEND_SAAS,
+            'url': saas_host,
+            'account_id': saas_account_id,
+            'database_id': backend_aware_saas_database_id,
+            'pat': saas_pat
+        }
+    return {}
+
+
+@pytest.fixture
+def backend_aware_database_params(backend,
+                                  backend_aware_onprem_database_params,
+                                  backend_aware_saas_database_params) -> dict[str, Any]:
+    """
+    Returns a set of parameters sufficient to open a pyexasol connection to the
+    current testing backend.
+    Usage example:
+    connection = pyexasol.connect(**backend_aware_database_params, compression=True)
+    """
+    if backend == _BACKEND_ONPREM:
+        return backend_aware_onprem_database_params
+    elif backend == _BACKEND_SAAS:
+        return backend_aware_saas_database_params
+    else:
+        ValueError(f'Unknown backend {backend}')
+
+
+@pytest.fixture
```

Comment:
Should this fixture be scope "session", too?

Reply:
The same answer, it is per-test by design.